### PR TITLE
fix(qwik-city): enforce / on base paths

### PR DESCRIPTION
**Pull request by wmertens**
@mhevery not sure if this should go in only v2 because it might be slightly breaking.

### CODE REVIEW

File: packages/qwik/src/server/utils.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ export function getBuildBase(opts: RenderToStringOptions) {
     }
     return base;
   }
-  return '/build/';
+  return `${import.meta.env.BASE_URL}build/`;
```
Comment (PatrickJS) [inline packages/qwik/src/server/utils.ts:30]:
```suggestion
  return `${import.meta.env.BASE_URL || '/'}build/`;
```

Comment (wmertens) [reply packages/qwik/src/server/utils.ts:30]: Isn't that always defined?

Comment (PatrickJS) [reply packages/qwik/src/server/utils.ts:30]: nope

Comment (wmertens) [reply packages/qwik/src/server/utils.ts:30]: So this [is determined by](https://vitejs.dev/guide/env-and-mode#:~:text=This%20is%20determined%20by%20the%20base%20config%20option.) [the base config option](https://vitejs.dev/config/shared-options.html#base) which defaults to `/`, and we check that it ends on a `/`, so I think it's always defined.

Comment (PatrickJS) [reply packages/qwik/src/server/utils.ts:30]: can we still do warning for now and throw errors later

**Comment by PatrickJS** [comment]
I think we might need q:origin because I want the base path to be the same but different origin. and if I want to have a certain version of base then I'll change the base path to that version in ssr entry

**Review by PatrickJS — APPROVED** [review]

**Review by gioboa — APPROVED** [review]

**Comment by wmertens** [comment]
@PatrickJS could you elaborate on origin? I don't understand the use case

**Comment by PatrickJS** [comment]
so the base path might be the same but you may change the origin to an assets domain. in one of my apps we're using base to change the origin to an assets cdn while the app is hosted on another origin domain

**Comment by wmertens** [comment]
@PatrickJS I think you can achieve it without q:origin. Set vite base to the URL prefix of the app and set the renderToStream base to the full url of the origin+prefix+build/

**Pull request merged**